R/tm_g_distribution.R

```diff
@@ -343,9 +336,89 @@ srv_distribution <- function(id,
   with_filter <- !missing(filter_panel_api) && inherits(filter_panel_api, "FilterPanelAPI")
   checkmate::assert_class(data, "tdata")
   moduleServer(id, function(input, output, session) {
-    data_extract <- list(dist_i = dist_var, strata_i = strata_var, group_i = group_var)
 
-    selector_list <- teal.transform::data_extract_multiple_srv(data_extract, data)
+    rule_req <- function(value) {
+      if (isTRUE(input$dist_tests %in% c("Fligner-Killeen",
+                                         "t-test (two-samples, not paired)",
+                                         "F-test",
+                                         "Kolmogorov-Smirnov (two-samples)",
+                                         "one-way ANOVA"))) {
+        if (!shinyvalidate::input_provided(value))
+          "Please select stratify variable."
+      }
+    }
+    rule_dupl <- function(...) {
+      if (identical(input$dist_tests, "Fligner-Killeen")) {
+        strata <- selector_list()$strata_i()$select
+        group <- selector_list()$group_i()$select
+        if (isTRUE(strata == group))
+          "Please select different variables for strata and group."
+      }
+    }
+
+    selector_list <- teal.transform::data_extract_multiple_srv(
+      data_extract = list(
+        dist_i = dist_var,
+        strata_i = strata_var,
+        group_i = group_var
+      ),
+      data,
+      select_validation_rule = list(
+        dist_i = shinyvalidate::sv_required("Please select a variable")
+      ),
+      filter_validation_rule = list(
+        strata_i = shinyvalidate::compose_rules(
+          rule_req,
+          rule_dupl
+        ),
+        group_i = rule_dupl
+      )
+    )
+
+    iv_r <- reactive({
```

Comment:
Not sure I completely get this - 
Would have thought this gives a validation as you've not selected any group by/strata by
![image](https://user-images.githubusercontent.com/15201933/210370414-fe821dd9-a81f-4d24-8a58-bf7e76f7ad69.png)

Reply:
Strata is only required in certain cases. These are data validations downstream of input.